tests/Doctrine/Tests/ORM/Utility/IdentifierFlattenerTest.php

```diff
@@ -87,7 +87,7 @@ public function testFlattenIdentifierWithOneToOneId()
 
         $this->assertArrayHasKey('secondEntity', $flatIds, 'It should be called secondEntity');
 
-        $this->assertSame($id['secondEntity']->id, $flatIds['secondEntity']);
+        $this->assertEquals($id['secondEntity']->id, $flatIds['secondEntity']);
```

Comment:
@goetas makes sense, though we really should stop doing that unless it's a hashed ID. I'll think about it, thanks.